components/pkg-export-helm/src/deps.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2018 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use failure::SyncFailure;
+use handlebars::Handlebars;
+use clap;
+use std::io::Write;
+use std::path::Path;
+use std::process::Command;
+
+use common::ui::{UI, Status};
+use export_docker::Result;
+
+pub const DEFAULT_OPERATOR_VERSION: &'static str = "0.5.1";
+pub const OPERATOR_REPO_URL: &'static str = "https://kinvolk.github.io/habitat-operator\
+                                             /helm/charts/stable/";
+
+// Helm requirements.yaml template
+const DEPSFILE: &'static str = include_str!("../defaults/HelmDeps.hbs");
+
+pub struct Deps {
+    operator_version: String,
+    update: bool,
+}
+
+impl Deps {
+    pub fn new_for_cli_matches(matches: &clap::ArgMatches) -> Self {
+        Deps {
+            operator_version: matches
+                .value_of("OPERATOR_VERSION")
+                .unwrap_or(DEFAULT_OPERATOR_VERSION)
+                .to_owned(),
+            update: matches.is_present("DOWNLOAD_DEPS"),
+        }
+    }
+
+    pub fn generate(&mut self, write: &mut Write) -> Result<()> {
+        let out = self.into_string()?;
+        write.write(out.as_bytes())?;
+
+        Ok(())
+    }
+
+    pub fn download<P: AsRef<Path>>(&self, dir: P, ui: &mut UI) -> Result<()> {
+        if !self.update {
+            return Ok(());
+        }
+        ui.status(Status::Downloading, "dependencies")?;
+
+        // Ignore the results cause we only install the repo as a courtesy to try to ensure that
+        // the actual `helm dep up` command afterwards succeeds. Currently `helm` doesn't mind if
+        // the repo is already added but we can't rely on that behaviour in future `helm` versions.
+        Command::new("helm")
```

Comment:
What does the error message look like if a user tries to run this without `helm` being present? Can the exporter check that early on and give an actionable error message to the user if it's not there, along with whatever setup instructions they might need to perform? (e.g. `helm init -c`, as you mention in your commit message)

Reply:
Also, the docker exporter uses a Habitat docker package; should we do something similar for Helm?

Reply:
I haven't checked but since `spawn` method returns an `std::io::Result`, I bet the error is of `std::io::ErrorKind::NotFound`. Regardless of the exact error, exporter can surely do that. :)

For the dependency, AFAICT helm already has a package but not in 'core' but rather 'inro' origin (no idea what that origin is about). Can we depend on that or do we need to move/create a new one to/for 'core'?

Reply:
So turns out that if the `helm` binary is not installed, we get error from the `spawn()` call but if it's not initialized, we don't get an error but can check the exit status of the child process from the return value of the wait() call. My upcoming patch handles these two cases separately to give helpful messages to user:

```
Ω Creating file `testapp-0.2.0-20180118130531/requirements.yaml`
✗✗✗
✗✗✗ Failed to launch `helm` command. Please ensure Helm is installed.
✗✗✗
```

```
Ω Creating file `testapp-0.2.0-20180118130531/requirements.yaml`
Error: open /home/zeenix/.helm/repository/repositories.yaml: no such file or directory
✗✗✗
✗✗✗ Failed to update chart dependencies. Please ensure Helm is initialized. For only setting up the Helm client, please run `helm init -c`.
✗✗✗
```

Reply:
I really like these error messages.

As for a helm package, I don't think `core` packages should be depending on non-`core` packages. We can look into a `core/helm` package as a separate issue; I think having these error messages will be very helpful in the meantime.